highfidelity/hifi
# Miscellaneous shader fixes

https://highfidelity.manuscript.com/f/cases/17425/Shader-issues

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/7bbb72ec23d3bcc1a9a2651ca5587d5c/raw/6570f92601072ad99116baaaa850f4bc3bc67cdf/ParabolaPointerTest.js).  A parabola should shoot out of your right hand.
- Confirm that particles render on Android (check: theSpot-dev)
- Run [this](https://gist.githubusercontent.com/SamGondelman/d2fe968f275be983ba2ed328437f1add/raw/012466e5f35091d0bab5ca36312f37d57775c883/CreatProcedurals.js).  You'll see this:
![samgondelman serverless_ file____ _serverless_tutorial json - build dev 8_7_2018 10_44_01 am](https://user-images.githubusercontent.com/7650116/43793249-d74e0ce2-9a2f-11e8-843d-f7a6e11dd882.png)
- Press 'u'.  You'll see this:
![samgondelman serverless_ file____ _serverless_tutorial json - build dev 8_7_2018 10_49_16 am](https://user-images.githubusercontent.com/7650116/43793250-d765c5b2-9a2f-11e8-8009-2ae920fc33fd.png)
- Press 'i'.  You'll see this:
![samgondelman serverless_ file____ _serverless_tutorial json - build dev 8_7_2018 10_49_25 am](https://user-images.githubusercontent.com/7650116/43793251-d77fb27e-9a2f-11e8-8d7d-3aeab4d57910.png)
- Press 'u'.  You'll see this:
![samgondelman serverless_ file____ _serverless_tutorial json - build dev 8_7_2018 10_50_25 am](https://user-images.githubusercontent.com/7650116/43793252-d795771c-9a2f-11e8-8d3c-96f0472f67c8.png)
- In a domain with dynamic entities, enable "Developer -> Physics -> Highlight Simulation Ownership".  You should see status icons around the dynamic entities, which should update when you bump into them.
- Open the running scripts dialog and search for debugCulling.js.  Run it.  Each of the checkboxes should have some effect (they trigger the drawing of various bounding boxes).  I've already confirmed that they draw the correct things, so this check is more that you don't crash or break rendering or anything,.
